### PR TITLE
[Testing] Ensure that "optional unowned" works at a basic level

### PR DESCRIPTION
**Pull request by davezarzycki**
This is followup to: https://github.com/apple/swift/pull/17767

### CODE REVIEW

**Comment by davezarzycki** [comment]
@swift-ci please test platform

File: packages/Python/lldbsuite/test/lang/swift/reference_storage_types/main.swift [hunk 1]
```diff
@@ -16,9 +16,12 @@ class Minion {
...
 class MyClass {
-    weak var sub_001: Minion! = Minion.TheMinionKing
-    var sub_002: Minion! = Minion.TheMinionKing
-    unowned(unsafe) var sub_003: Minion = Minion.TheMinionKing
+    /*default*/     var sub_001: Minion! = Minion.TheMinionKing
+    weak            var sub_002: Minion! = Minion.TheMinionKing
+    unowned         var sub_003: Minion! = Minion.TheMinionKing
+    unowned(unsafe) var sub_004: Minion! = Minion.TheMinionKing
+    unowned         var sub_005: Minion  = Minion.TheMinionKing
+    unowned(unsafe) var sub_005: Minion  = Minion.TheMinionKing
```
Comment (jimingham) [inline packages/Python/lldbsuite/test/lang/swift/reference_storage_types/main.swift]: Did you mean this last one to be sub_006?

**Review by jimingham — CHANGES_REQUESTED** [review]
There's a typo in the main.swift file.  Other than that this looks fine.

**Comment by jrose-apple** [comment]
I don't want to hold up this PR, but the case I'm worried about is when the `unowned` property is `nil`.

**Comment by davezarzycki** [comment]
@jimingham – I've made the requested fix.

@jrose-apple, @dcci, and @jimingham – Please see this bug as a follow up: https://bugs.swift.org/browse/SR-8285

@swift-ci please test platform

**Comment by davezarzycki** [comment]
@swift-ci please test platform

**Review by jimingham — APPROVED** [review]
What's here looks fine.  I thought Jordan was asking for another test for a nil unowned property.  I have an independent report of problems with the Linux REPL right now, but you should be able to add the same property to the test here...

**Comment by davezarzycki** [comment]
Fixed another copy-and-paste error.

@swift-ci please test platform

**Review by dcci — APPROVED** [review]
lgtm.
